cmake/onednn.cmake

```diff
@@ -1,27 +1,25 @@
 include_guard()
+include(functions)
 
-set(ONEDNN_CMAKE_ARGS
-        -DDNNL_IS_MAIN_PROJECT=FALSE
-        -DDNNL_BUILD_TESTS=FALSE
-        -DDNNL_BUILD_EXAMPLES=FALSE
+gc_fetch_content(dnnl main https://github.com/intel-ai/oneDNN.git
+        SKIP_ADD
+        CMAKE_ARGS -DDNNL_IS_MAIN_PROJECT=FALSE -DDNNL_BUILD_TESTS=FALSE -DDNNL_BUILD_EXAMPLES=FALSE
 )
 
-if (${GC_ONEDNN_VERSION} STREQUAL "main")
-    set(ONEDNN_TAG ${GC_ONEDNN_VERSION})
-else ()
-    set(ONEDNN_TAG v${GC_ONEDNN_VERSION})
+# This allows to generate headers from *.in without adding the library to the build.
+# If the build is required, remove this and the SKIP_ADD option above.
+if (DEFINED CMAKE_GENERATOR)
+    set(GENERATOR_FLAG "-G ${CMAKE_GENERATOR}")
 endif ()
-message(STATUS "oneDNN version: ${GC_ONEDNN_VERSION}")
-set(FETCHCONTENT_QUIET FALSE)
-FetchContent_Declare(
-        oneDNN
-        GIT_REPOSITORY https://github.com/intel-ai/oneDNN.git
-        GIT_TAG ${ONEDNN_TAG}
-        CMAKE_ARGS ${ONEDNN_CMAKE_ARGS}
-        GIT_PROGRESS TRUE
+execute_process(COMMAND ${CMAKE_COMMAND} ${GENERATOR_FLAG}
+        -Wno-dev
+        -S ${dnnl_SOURCE_DIR}
+        -B ${dnnl_BINARY_DIR}
+        ${GC_DNNL_CMAKE_ARGS}
 )
 
-FetchContent_Populate(oneDNN)
-FetchContent_GetProperties(oneDNN)
-message(STATUS "oneDNN source dir: ${onednn_SOURCE_DIR}")
-FetchContent_MakeAvailable(oneDNN)
\ No newline at end of file
+include_directories(
+        ${dnnl_BINARY_DIR}/include
+        ${dnnl_SOURCE_DIR}/include
+        ${dnnl_SOURCE_DIR}/src/graph/backend/elyzor/include
+)
```

Comment:
We should probably export some `${name}_INCLUDES` and then use it properly in `target_include_directories`.

Reply:
Now  the script sets the DNNL_INCLUDES variable.